File: mobile/flutter/android/flutter-bundle/src/main/java/com/smart/library/flutter/utils/FlutterRouter.java
package com.smart.library.flutter.utils;

import android.content.Context;
import android.content.Intent;

import com.smart.library.flutter.views.FlutterOrderActivity;
import com.smart.library.flutter.views.FlutterSettingsActivity;
import com.smart.library.flutter.views.NativeMineActivity;

@SuppressWarnings({"WeakerAccess", "UnusedReturnValue"})
public class FlutterRouter {
    public static final String URL_MINE = "flutter://native/mine";
    public static final String URL_ORDER = "flutter://flutter/order";
    public static final String URL_SETTINGS = "flutter://flutter/settings";

    public static boolean openPageByUrl(Context context, String url) {
        return openPageByUrl(context, url, 0);
    }

    public static boolean openPageByUrl(Context context, String url, int requestCode) {
        try {
            if (url.startsWith(URL_ORDER)) {
                context.startActivity(new Intent(context, FlutterOrderActivity.class));
                return true;
            } else if (url.startsWith(URL_SETTINGS)) {
                context.startActivity(new Intent(context, FlutterSettingsActivity.class));
                return true;
            } else if (url.startsWith(URL_MINE)) {
                context.startActivity(new Intent(context, NativeMineActivity.class));
                return true;
            } else {
                return false;
            }
        } catch (Throwable t) {
            return false;
        }
    }
}

File: mobile/flutter/android/app/src/main/java/com/smart/library/flutter/FlutterRouter.kt
package com.smart.library.flutter

import android.app.Activity
import android.content.Context
import com.smart.library.flutter.test.MineActivity

@Suppress("MemberVisibilityCanBePrivate", "unused")
enum class FlutterRouter(val url: String, val goTo: (context: Context?, containerParams: MutableMap<String, Any>?, requestCode: Int, exts: MutableMap<String, Any>?) -> Boolean) {

    URL_MINE("flutter://native/mine", { context: Context?, containerParams: MutableMap<String, Any>?, requestCode: Int, exts: MutableMap<String, Any>? ->
        context?.startActivity(android.content.Intent(context, MineActivity::class.java))
        false
    }),
    URL_ORDER("flutter://flutter/order", { context: Context?, containerParams: MutableMap<String, Any>?, requestCode: Int, exts: MutableMap<String, Any>? ->
        FlutterActivity.startForResult(context as? Activity, URL_ORDER.url, containerParams, requestCode)
        true
    }),
    URL_SETTINGS("flutter://flutter/settings", { context: Context?, containerParams: MutableMap<String, Any>?, requestCode: Int, exts: MutableMap<String, Any>? ->
        FlutterActivity.startForResult(context as? Activity, URL_SETTINGS.url, containerParams, requestCode)
        true
    });

    companion object {
        fun find(url: String?): FlutterRouter? {
            return try {
                values().find { it.url == url }
            } catch (e: Exception) {
                null
            }
        }

        fun openContainer(context: Context?, url: String?, urlParams: MutableMap<String, Any>?, requestCode: Int, exts: MutableMap<String, Any>?) {
            find(url)?.goTo?.invoke(context, urlParams, requestCode, exts)
        }

    }

}